spec/registry/unicode.md

```diff
@@ -0,0 +1,63 @@
+# MessageFormat 2.0 Unicode Registry
+
+The `u:` namespace is reserved for use by the Unicode Consortium.
+
+## Options
+
+This section describes common options which each implementation SHOULD support
+for all _functions_ and _markup_.
+
+### `u:id`
+
+A string value that is included as an `id` or other suitable value
+in the formatted parts for the _placeholder_,
+or any other structured formatted results.
+
+Ignored when formatting a message to a string.
+
+Accepts string values, or values which can be stringified without error.
+For other values, a _Bad Option_ error is emitted
+and the `u:id` option is ignored.
+
+### `u:locale`
+
+A comma-delimited list of BCP 47 language tags,
+or an implementation-defined list of such tags.
```

Comment:
Perhaps (because ULI's are a specific narrowing of the validity rules and we are, after all, the CLDR-TC):

```suggestion
A comma-delimited prioritized list of BCP47 language tags
which are expected to be Unicode Locale Identifiers,
or an implementation-defined list of locale identifiers.
```

Reply:
Unfortunately, ULI is not only a narrowing of the validity rules, it also explicitly allows for `_` as a separator, so e.g. `en_US` is a valid ULI but not a valid BCP47 tag. This is not universal, see e.g. tc39/ecma402#777 for an open discussion of supporting this in JS.

Reply:
It needs stronger language than "which are expected to be".

```suggestion
A comma-delimited prioritized list of BCP47 language tags that are also [valid Unicode CLDR locale identifiers](https://www.unicode.org/reports/tr35/#BCP_47_Conformance).
```

I think allowing "implementation-defined locale identifiers" is dangerous, and will lead to interoperability problems. What is the thinking behind allowing these? `locale:|The King's English|`